src/PrecedenceGraph.h

```diff
@@ -109,6 +109,32 @@ class RecursiveClauses : public AstAnalysis {
     bool computeIsRecursive(const AstClause& clause, const AstTranslationUnit& translationUnit) const;
 };
 
+/**
+ * Analysis pass mapping identifiers with relations and clauses.
+ */
+class RelationDetail : public AstAnalysis {
+public:
+    static constexpr const char* name = "relation-detail";
+
+    RelationDetail() : AstAnalysis(name) {}
+
+    void run(const AstTranslationUnit& translationUnit) override;
+
+    void print(std::ostream& os) const override;
+
+    AstRelation* getRelation(const AstQualifiedName& name) const {
+        return nameToRelation.at(name);
```

Comment:
This is throwing errors in semantic/agg_checks2 as a name is not found. The name is for an undefined relation.